optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStream.java

```diff
@@ -685,6 +685,106 @@
                     BiConstraintCollector<A, B, ResultContainerC_, ResultC_> collectorC,
                     BiConstraintCollector<A, B, ResultContainerD_, ResultD_> collectorD);
 
+    /**
+     * Convert the {@link BiConstraintStream} to a {@link TriConstraintStream}, consisting of unique tuples with three
+     * facts.
+     * <p>
+     * The first fact is the return value of the first group key mapping function, applied on the incoming tuple.
+     * The second fact is the return value of the second group key mapping function, applied on all incoming tuples with
+     * the same first fact.
```

Comment:
"the first fact is ..." "the second fact is  ... with the same first fact". That "same first fact" seems wrong. Or at least confusing.
The second result fact isn't more dependent on the first result fact than visa versa, this line seems to imply that.

I'd scratch this paragraph with 3 lines and just add an example instead that shows that illustrates when and when not tuples get grouped together.